src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete state[keyToRemove];
+        }
+        break;
+      case 'clear':
+        for (const stateKey of Object.keys(state)) {
+          delete state[stateKey];
+        }
+        break;
+      default:
+        throw new Error('Bad state.');
+    }
+  }
```

Comment:
For readability add spaces:
And I think this error message would be more suitable
```suggestion
  for (const action of actions) {
    switch (action.type) {
      case 'addProperties':
        Object.assign(state, action.extraData);
        break;
        
      case 'removeProperties':
        for (const keyToRemove of action.keysToRemove) {
          delete state[keyToRemove];
        }
        break;
        
      case 'clear':
        for (const stateKey of Object.keys(state)) {
          delete state[stateKey];
        }
        break;
        
      default:
        throw new Error('Incorrect action type');
    }
  }
```

Reply:
I added new lines between cases and more descriptive error message.